Steps 2 GitOps tutorials (#2322)
giantswarm/docs: src/content/tutorials/fleet-management/app-platform/app-bundle/index.md

</h5>
No, at least not in the current implementation form. The reason is, app bundles install `App` custom resources that are understood only by the management clusters. Attempt to install them directly to the workload cluster would result in an error because corresponding `App` custom resource definition isn't available there, resulting in the `kubernetes` not understanding the submitted objects.
</div>

Now, obviously in order to leverage this management cluster layer, the bundle `App` custom resource must be configured in a certain way which is slightly different in comparison to the solitary apps. Find more details in the [Technical Details paragraph](#technical-details).

### Technical details {#technical-details}

#### Naming the bundle's `App` resource

The first, not so obvious, consequence of installing a bundle in a management cluster before its nested apps are installed in the workload cluster, is the requirement for name uniqueness. When a given app bundle is to be installed for more than one workload cluster, each `App` custom resource instance of this bundle must be given a unique name within the management cluster.

> It comes natural for the solitary apps, say the [`Hello World` app](https://github.com/giantswarm/hello-world-app),when one needs to install it twice in the workload cluster, both `App` custom resources instances must be named uniquely. It's natural because both live under the same namespace, but even if we could break them into separate namespaces, this uniqueness would still be demanded because both apps targets the same workload cluster, hence must be distinguished by its operators.

Reviewer: This doesn't read well at all ...

```suggestion
It becomes natural for solitary apps, for example, the [`Hello World` app](https://github.com/giantswarm/hello-world-app), that when one needs to install it twice in the workload cluster, both `App` custom resources instances must be named uniquely. It's natural because both live under the same namespace, but even if we could break them into separate namespaces, this uniqueness would still be demanded because both apps targets the same workload cluster, hence must be distinguished by its operators.
```

It's less obvious for the bundle apps, because these are usually scattered across the namespaces for the corresponding workload clusters from the very beginning, hence giving the impression they're isolated, which we already know to not be the case. Not complying to this rule results in apps competing over certain resources, and hence conflicting.

> Let's walk through an example to understand it better. Let's imagine a user has two workload clusters created, the `WC1` and the `WC2`, on the management cluster. The user then installs, via the `App` custom resource called just `bundle`, an app bundle in the `WC1` cluster. Everything works perfectly so far, see the figure below.

Reviewer: ```suggestion
Let's walk through an example to understand it better.

Imagine that a user has two workload clusters created, the `WC1` and the `WC2`, on the management cluster. The user then installs, via the `App` custom resource called just `bundle`, an app bundle in the `WC1` cluster. Everything works perfectly so far, see the figure below.
```

![User installs a bundle once](app-bundle-naming-conflict-step-1.png)

> The flow of actions can be thought of as an unbroken link. The continuity of this link is guarantee of delivering end resources. Now let's assume user wants to install the same app bundle, but for the `WC2` cluster this time.

Reviewer: ```suggestion
The flow of actions can be thought of as an unbroken link. The continuity of this link is guarantee of delivering end resources. Now let's assume the user wants to install the same app bundle, but for the `WC2` cluster this time.
```
